usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,743 @@
+            HMD.displayModeChanged.connect(_this.displayModeChanged);
+            Window.domainChanged.connect(_this.domainChanged);
+
+            _this.registerControllerMapping();
+            if (HMD.active) {
+                HMD.closeTablet();
+                _this.setUpHMDMode();
+            } else {
+                _this.setUpDesktopMode();
+            }
+            _this.playSound(OPEN_SOUND, OPEN_SOUND_VOLUME, MyAvatar.position, true, false);
+        },
+
+        /* PLAY A SOUND: Plays the specified sound at the position of the user's Avatar using the volume and playback 
+        mode requested. */
+        playSound: function(sound, volume, position, localOnly, loop){
+            if (sound.downloaded) {
+                if (injector) {
+                    injector.stop();
+                    injector = null;
+                }
+                injector = Audio.playSound(sound, {
+                    position: position,
+                    volume: volume,
+                    localOnly: localOnly,
+                    loop: loop
+                });
+            }
+        },
+
+        findWhiteboard: function() {
+            Entities.findEntities(MyAvatar.position, WHITEBOARD_SEARCH_RADIUS_M).forEach(function(entity) {
+                var properties = Entities.getEntityProperties(entity, ['position', "name"]);
+                if (properties.name && properties.name === "Whiteboard") {
+                    if (whiteboard) {
+                        if (Vec3.distance(properties.position, MyAvatar.position) <
+                            Vec3.distance(Entities.getEntityProperties(whiteboard, "position").position, MyAvatar.position)) {
+                            whiteboard = entity;
+                        }
+                    } else {
+                        whiteboard = entity;
+                    }
+                }
+                if (whiteboard) {
+                    whiteboardParts = Entities.getChildrenIDs(whiteboard);
+                    whiteboardParts.push(whiteboard);
+                }
+            });
+        },
+
+        draw: function(onBoard) {
+            if (!readyToDraw) {
+                return;
+            }
+            if (Vec3.distance(previousLinePoint, currentPoint) < MINIMUM_MOVEMENT_TO_DRAW_M ||
+            Vec3.distance(previousLinePoint, currentPoint) > MAXIMUM_MOVEMENT_TO_DRAW_M) {
+                return;
+            }
+            if (onBoard !== wasLastPointOnBoard) { // toggle between on board and air, stop drawing
+                _this.stopDrawing();
+                return;
+            }
+            wasLastPointOnBoard = onBoard;
+            var newLine = !polyLine;
+            var lineProperties = DEFAULT_LINE_PROPERTIES;
+            var linePointsCount;
+            if (!newLine) { // maybe editing existing line
+                var previousLineProperties = Entities.getEntityProperties(polyLine, ['linePoints', 'normals', 
```

Comment:
what happens after this if `polyLine` isn't a valid entity?

Reply:
That has never happened. Are you saying we need to check the validity of every property we are requesting?

Reply:
I suppose it could happen if 2 people are using the board and one is drawing a line while the other hits reset.

Reply:
yeah, that's one case i could see that happening. or not just reset - if someone else clears the board using any method (even Create) while someone else is drawing, there'd be a problem)

yes, you should check that the properties you expect exist before acting on the data. something like this would work:
```
if (!(previousLineProperties.linePoints && previousLineProperties.normals && previousLineProperties.strokeWidths && previousLineProperties.age)) {
   <error handling>
   return;
}
```

Reply:
got it..thanks!